packages/@aws-cdk/cdk/lib/app.ts

```diff
@@ -125,27 +188,25 @@ export class App extends Root {
     }
   }
 
-  private collectRuntimeInformation(): cxapi.AppRuntime {
-    const libraries: { [name: string]: string } = {};
+  protected prepare() {
+    this.prepared = true;
+  }
 
-    for (const fileName of Object.keys(require.cache)) {
-      const pkg = findNpmPackage(fileName);
-      if (pkg && !pkg.private) {
-        libraries[pkg.name] = pkg.version;
-      }
+  protected validate(): string[] {
+    this.validated = true;
+    if (numberOfAppsInProgram(this) > 1 && this.node.id === DEFAULT_APP_NAME) {
+      return ['When constructing more than one App, all of them must have ids'];
     }
+    return [];
+  }
 
-    // include only libraries that are in the @aws-cdk npm scope
-    for (const name of Object.keys(libraries)) {
-      if (!name.startsWith('@aws-cdk/')) {
-        delete libraries[name];
-      }
-    }
-
-    // add jsii runtime version
-    libraries['jsii-runtime'] = getJsiiAgentVersion();
-
-    return { libraries };
+  /**
+   * Synthesize the App
+   */
+  protected synthesize(): any {
```

Comment:
Should skip the `: any` here. I reckon the inferred type will still be better info than that.